LOGGING/src/NLogTarget/ApplicationInsightsTarget.cs

```diff
@@ -118,10 +129,24 @@ protected override void InitializeTarget()
             this.telemetryClient = new TelemetryClient();
 #pragma warning restore CS0618 // Type or member is obsolete
 
-            string instrumentationKey = this.instrumentationKeyLayout.Render(LogEventInfo.CreateNullEvent());
-            if (!string.IsNullOrWhiteSpace(instrumentationKey))
+            string connectionString = this.connectionStringLayout.Render(LogEventInfo.CreateNullEvent());
+           
+            // Check if nlog application insights target has connectionstring in config file then
+            // configure telemetryclient with the connectionstring otherwise using instrumentationkey.
+            if (!string.IsNullOrWhiteSpace(connectionString))
             {
-                this.telemetryClient.Context.InstrumentationKey = instrumentationKey;
+                var configuration = TelemetryConfiguration.CreateDefault();
```

Comment:
why create a new config here, and that seems ununsed?

Reply:
@cijothomas  - We have removed unnecessary config creation.